django/db/migrations/operations/models.py

```diff
@@ -277,70 +278,50 @@ def deconstruct(self):
             kwargs
         )
 
+    def _get_model_key(self, remote_model, app_label, model_name):
+        if remote_model == RECURSIVE_RELATIONSHIP_CONSTANT:
+            return app_label, model_name.lower()
+        elif '.' in remote_model:
+            return tuple(remote_model.lower().split('.'))
+        else:
+            return app_label, remote_model.lower()
+
     def state_forwards(self, app_label, state):
-        # In cases where state doesn't have rendered apps, prevent subsequent
-        # reload_model() calls from rendering models for performance
-        # reasons. This method should be refactored to avoid relying on
-        # state.apps (#27310).
-        reset_apps = 'apps' not in state.__dict__
-        apps = state.apps
-        model = apps.get_model(app_label, self.old_name)
-        model._meta.apps = apps
-        # Get all of the related objects we need to repoint
-        all_related_objects = (
-            f for f in model._meta.get_fields(include_hidden=True)
-            if f.auto_created and not f.concrete and (not f.hidden or f.many_to_many)
-        )
-        if reset_apps:
-            del state.__dict__['apps']
-        # Rename the model
-        state.models[app_label, self.new_name_lower] = state.models[app_label, self.old_name_lower]
-        state.models[app_label, self.new_name_lower].name = self.new_name
+        # Add a new model.
+        renamed_model = state.models[app_label, self.old_name_lower].clone()
+        renamed_model.name = self.new_name
+        state.models[app_label, self.new_name_lower] = renamed_model
+        # Repoint all fields pointing to the old model to the new one.
+        old_model_key = app_label, self.old_name_lower
+        new_remote_model = "%s.%s" % (app_label, self.new_name)
+        for (model_app_label, model_name), model_state in state.models.items():
+            model_changed = False
+            for index, (name, field) in enumerate(model_state.fields):
+                changed_field = None
+                remote_field = field.remote_field
+                if remote_field:
+                    remote_model_key = self._get_model_key(
+                        remote_field.model, model_app_label, model_name
+                    )
+                    if remote_model_key == old_model_key:
+                        changed_field = field.clone()
+                        changed_field.remote_field.model = new_remote_model
+                    through_model = getattr(remote_field, 'through', None)
```

Comment:
I'm afraid it's not the case as `ForeignKey` and `OneToOneField` don't have such an attribute.